### ISSUE TITLE
[a11y]: Site hamburger menu (appears when zoomed 200%) does not hide background content from JAWS and other assisteive technology

### ISSUE THREAD

**Issue by philljenkins** [carbon-design-system/carbon#14138]
### Package

@carbon/react

### Browser

Chrome

### Operating System

Windows

### Package version

v11

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker 3.1.52

### Assistive technology

JAWS 2023

### Description

while hamburger menu is opened, focus is moving out and elements are getting truncated 

### Reproduction/example

WCAG 2.1 violation: 1.4.4 Resize text

### Steps to reproduce

1. Load JAWS
2. Load ibm.com/able
3. Zoom to 200%
4. Tab to Site hamburger menu that recently appeared due to zooming and reflow
5. Tab to items in the menu,
6. continue tabbing incorrectly to items outside of the menu

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by philljenkins** [comment]
Original issue was opened against the IBM Accessibility toolkit https://github.ibm.com/IBMa/E2E/issues/4795
